File: src/main/java/com/yolonews/YoloNewsApp.java
package com.yolonews;

import com.google.inject.Guice;
import com.google.inject.Injector;
import com.yolonews.auth.AuthAPI;
import com.yolonews.auth.AuthFilter;
import com.yolonews.auth.SecurityContextFilter;
import com.yolonews.auth.UserAPI;
import com.yolonews.posts.PostAPI;
import io.dropwizard.Application;
import io.dropwizard.setup.Environment;

/**
 * @author saket.mehta
 */
class YoloNewsApp extends Application<YoloNewsConfig> {
    public static void main(String[] args) throws Exception {
        new YoloNewsApp().run(args);
    }

    @Override
    public void run(YoloNewsConfig configuration, Environment environment) {
        Injector injector = Guice.createInjector(new YoloNewsModule());

        // api
        environment.jersey().register(injector.getInstance(AuthAPI.class));
        environment.jersey().register(injector.getInstance(UserAPI.class));
        environment.jersey().register(injector.getInstance(PostAPI.class));

        // filters
        environment.jersey().register(injector.getInstance(SecurityContextFilter.class));
        environment.jersey().register(injector.getInstance(AuthFilter.class));
    }
}

File: src/main/kotlin/yolonews/YOLONewsApp.kt
package yolonews

import com.fasterxml.jackson.module.kotlin.KotlinModule
import io.dropwizard.Application
import io.dropwizard.setup.Bootstrap
import io.dropwizard.setup.Environment
import redis.clients.jedis.JedisPool
import yolonews.api.AuthAPI
import yolonews.api.UserAPI
import yolonews.services.AuthService
import yolonews.services.UserService


/**
 * @author saket.mehta
 */
class YOLONewsApp : Application<YOLONewsConfig>() {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            YOLONewsApp().run(*args)
        }
    }

    override fun run(config: YOLONewsConfig, environment: Environment) {
        // Initialise Redis
        val jedisPool = JedisPool(config.redisURL)

        // Resources
        environment.jersey().register(AuthAPI(AuthService(jedisPool)))
        environment.jersey().register(UserAPI(UserService(jedisPool)))

        // Filters
        val authFilter = AuthFilter(config.tokenHeaderPrefix!!, config.tokenQueryParamKey!!, AuthService(jedisPool))
        environment.jersey().register(authFilter)
    }

    override fun initialize(bootstrap: Bootstrap<YOLONewsConfig>) {
        bootstrap.objectMapper.registerModule(KotlinModule())
    }
}